src/lib.rs

```diff
@@ -209,9 +209,49 @@ impl<'a> ParseOptions<'a> {
         self
     }
 
-    /// Call the provided function or closure on non-fatal parse errors.
+    /// Call the provided function or closure on non-fatal parse errors, passing
+    /// a static string description.  This method is deprecated in favor of
+    /// `syntax_violation_callback` and is implemented as an adaptor for the
+    /// latter, passing the `SyntaxViolation` description. Only the last value
+    /// passed to either method will be used by a parser.
+    #[deprecated]
```

Comment:
I'm not sure the method should immediately be made deprecated.

Reply:
Its not required for my use case.   But why not give deprecation warning to those (relatively few) users of this feature to migrate to Fn(SyntaxViolation), calling  `v.description()` themselves if that is all they want? If deprecated in the next minor release (merged), then this could be removed in some future major release, as cleanup.